src/main/java/seedu/address/logic/commands/edit/EditCommand.java

```diff
@@ -90,8 +91,15 @@ public CommandResult execute(Model model) throws CommandException {
         Person personToEdit = lastShownList.get(index.getZeroBased());
         Person editedPerson = createEditedPerson(personToEdit, editPersonDescriptor);
 
-        if (!personToEdit.isSamePerson(editedPerson) && model.hasPerson(editedPerson)) {
+        boolean phoneExists = model.hasPhone(editedPerson);
+        boolean emailExists = model.hasEmail(editedPerson);
+
+        if (!personToEdit.isSamePerson(editedPerson) && phoneExists && emailExists) {
             throw new CommandException(MESSAGE_DUPLICATE_PERSON);
```

Comment:
hmm actually maybe we can skip this check? because i think saying "this person already exists" is less meaningful than telling them that the phone number already exists or the email already exists